tests/integration/conftest.py

```diff
@@ -634,37 +641,99 @@ def installation_ctx(  # pylint: disable=too-many-arguments
+    tables[orc_table_name] = context.make_table(
+        schema_name=schema.name,
+        name=orc_table_name,
+        hiveserde_ddl=orc_ddl,
+        storage_override=f"{table_base_dir}/{orc_table_name}",
+    )
+
+    avro_table_name = f"avro_serde_{random}"
+    avro_ddl = f"""CREATE TABLE hive_metastore.{schema.name}.{avro_table_name} (id INT, region STRING) 
+                        ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.avro.AvroSerDe'
+                        STORED AS INPUTFORMAT 'org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat'
+                                  OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat'
+                        TBLPROPERTIES ('avro.schema.literal'='{{
+                            "namespace": "org.apache.hive", 
+                            "name": "first_schema", 
+                            "type": "record",
+                            "fields": [
+                                {{ "name":"id", "type":"int" }},
+                                {{ "name":"region", "type":"string" }}
+                            ] }}') 
+                        LOCATION '{table_base_dir}/{avro_table_name}'
+                    """
+    tables[avro_table_name] = context.make_table(
+        schema_name=schema.name,
+        name=avro_table_name,
+        hiveserde_ddl=avro_ddl,
+        storage_override=f"{table_base_dir}/{avro_table_name}",
+    )
+    return tables
+
+
+def prepare_regular_tables(context, external_csv, schema) -> dict[str, TableInfo]:
     tables: dict[str, TableInfo] = {
-        "src_managed_table": installation_ctx.make_table(schema_name=schema.name),
-        "src_external_table": installation_ctx.make_table(schema_name=schema.name, external_csv=make_mounted_location),
+        "src_managed_table": context.make_table(schema_name=schema.name),
+        "src_external_table": context.make_table(schema_name=schema.name, external_csv=external_csv),
     }
     src_view1_text = f"SELECT * FROM {tables['src_managed_table'].full_name}"
-    tables["src_view1"] = installation_ctx.make_table(
+    tables["src_view1"] = context.make_table(
         catalog_name=schema.catalog_name,
         schema_name=schema.name,
         ctas=src_view1_text,
         view=True,
     )
     src_view2_text = f"SELECT * FROM {tables['src_view1'].full_name}"
-    tables["src_view2"] = installation_ctx.make_table(
+    tables["src_view2"] = context.make_table(
         catalog_name=schema.catalog_name,
         schema_name=schema.name,
         ctas=src_view2_text,
         view=True,
     )
+    return tables
+
+
+@pytest.fixture
+def prepare_tables_for_migration(
+    ws, installation_ctx, make_catalog, make_random, make_mounted_location, env_or_skip, make_storage_dir, request
+) -> tuple[dict[str, TableInfo], SchemaInfo]:
+    scenario = request.param
```

Comment:
okay, i see what you're doing here - please add a comment how it works with links to pytest documentation. I was not familiar with the parametrised calls of a fixture. There's even less chance of other people being familiar with it as well. can you return `Callable[[str], tuple[dict[str, TableInfo], SchemaInfo]]` - it's more aligned with how all `make_*` fixtures are working.

Reply:
Added code comment with pytest document link to explain the `request.param` here. 
Yes, this function can be rewrite using the inner functions like `make_*` and thus we don't need this `request.param`. But I will address this in a separate PR.